src/hotspot/share/opto/predicates.cpp

```diff
@@ -1093,11 +1094,18 @@ CloneUnswitchedLoopPredicatesVisitor::CloneUnswitchedLoopPredicatesVisitor(
     PhaseIdealLoop* phase)
     : _clone_predicate_to_true_path_loop(true_path_loop_head, node_in_true_path_loop_body, phase),
       _clone_predicate_to_false_path_loop(false_path_loop_head, node_in_false_path_loop_body, phase),
-      _phase(phase) {}
+      _phase(phase),
+      _is_counted_loop(true_path_loop_head->is_CountedLoop()) {}
 
-// Keep track of whether we are in the correct Predicate Block where Template Assertion Predicates can be found.
 // The PredicateIterator will always start at the loop entry and first visits the Loop Limit Check Predicate Block.
+// Does not clone Loop Limit Check parse predicates iff a counted loop is unswitched, because a loop limit check before
+// the unswitched loop selector covers both unswitched counted loops. Otherwise, we would need to hoist the loop limit
+// checks from both loops back up to the loop selector.
```

Comment:
I suggest the following rephrasing to highlight why cloning the Parse Predicate is not useful (note that Loop Unswitching does not clone the actual Loop Limit Check but rather the Parse Predicate which is a placeholder for a potential Loop Limit Check later).
```suggestion
// Does not clone a Loop Limit Check Parse Predicate if a counted loop is unswitched, because it most likely will not be used anymore (it could only be used when both unswitched loop versions die and the Loop Limit Check Parse Predicate ends up at a LoopNode without Loop Limit Check Parse Predicate directly following the unswitched loop that can then be speculatively converted to a counted loop - this is rather rare).
```